src/IIIFPresentation/API/Features/Storage/StorageController.cs

```diff
@@ -42,6 +42,20 @@ public async Task<IActionResult> GetHierarchicalCollection(int customerId, strin
                 ContentTypes.V3)
             : Content(storageRoot.StoredCollection, ContentTypes.V3);
     }
+    
+    [HttpPost("{*slug}")]
+    [ETagCaching]
+    public async Task<IActionResult> PostHierarchicalCollection(int customerId, string slug)
+    {
+        if (!Request.ShowExtraProperties())
+        {
+            return this.PresentationProblem(statusCode: (int)HttpStatusCode.Forbidden);
+        }
+        
+        var rawRequestBody = await new StreamReader(Request.Body).ReadToEndAsync();
```

Comment:
Does `StreamReader` need to be disposed of? Or would that close `Request.Body`? and if it did is that bad?